packages/admin/admin/src/common/FieldSet.tsx

```diff
@@ -0,0 +1,147 @@
+            lineHeight: "18px",
+            color: theme.palette.text.secondary,
+        },
+        endAdornment: { display: "flex", alignItems: "center" },
+        placeholder: {
+            flexGrow: 1,
+            boxSizing: "inherit",
+            userSelect: "none",
+        },
+        children: {
+            display: "flex",
+            flexDirection: "column",
+            borderTop: `solid ${theme.palette.divider}`,
+            padding: "20px",
+            [theme.breakpoints.up("sm")]: {
+                padding: "40px",
+            },
+            "&$disablePadding": {
+                padding: "0px",
+            },
+        },
+        disablePadding: {},
+    });
+
+function FieldSet({
+    title,
+    supportText,
+    endAdornment,
+    children,
+    collapsible = true,
+    initiallyExpanded = true,
+    disablePadding = false,
+    componentsProps,
+    classes,
+}: React.PropsWithChildren<FieldSetProps> & WithStyles<typeof styles>): React.ReactElement {
+    const [expanded, setExpanded] = React.useState(initiallyExpanded);
+
+    const handleChange = (event: React.SyntheticEvent, isExpanded: boolean) => {
+        setExpanded(isExpanded);
+    };
+
+    return (
+        <MuiAccordion
+            expanded={expanded}
+            onChange={
+                collapsible
+                    ? handleChange
+                    : () => {
+                          /* do nothing */
+                      }
+            }
+        >
+            <MuiAccordionSummary
+                classes={{ root: classes.header }}
+                expandIcon={collapsible && <ArrowForwardIosSharpIcon />}
+                {...componentsProps?.fieldSetSummary}
+            >
+                <div className={clsx(classes.headerColumn)}>
+                    <div className={clsx(classes.title, expanded && classes.disablePadding)}>{title}</div>
+                    <div className={clsx(classes.supportText)}>{supportText}</div>
+                </div>
+                <div className={clsx(classes.placeholder)} />
+                <div className={clsx(classes.endAdornment)}>{endAdornment}</div>
+            </MuiAccordionSummary>
+            <MuiAccordionDetails className={clsx(classes.children, disablePadding && classes.disablePadding)} {...componentsProps?.fieldSetDetails}>
+                {children}
+            </MuiAccordionDetails>
+        </MuiAccordion>
+    );
+}
+
+const FieldSetWithStyles = withStyles(styles, { name: "CometAdminFieldSet" })(FieldSet);
+
+export { FieldSetWithStyles as FieldSet };
+
+declare module "@mui/material/styles" {
+    interface ComponentsPropsList {
+        CometAdminFieldSet: FieldSetProps;
+    }
+
+    interface ComponentNameToClassKey {
+        CometAdminFieldSet: FieldSetClassKey;
+    }
+
+    interface Components {
+        CometAdminFieldSet?: {
+            defaultProps?: ComponentsPropsList["CometAdminFieldSet"];
+            styleOverrides?: ComponentNameToClassKey["CometAdminFieldSet"];
```

Comment:
This should use the `ComponentsOverrides` type with the `Theme` type as a generic.  
Sorry, this is currently incorrect in the docs, will be changed soon. 

```suggestion
            styleOverrides?: ComponentsOverrides<Theme>["CometAdminFieldSet"];
```